apps/web/src/pages/api/v1/admin/update-claims.ts

```diff
@@ -40,12 +27,25 @@ handler.patch(
     if (!request.body.role) {
       throw new BadRequest('No role provided')
     }
+
+    // Check permissions (not using admin middleware)
     const admin = configureAdmin()
-    const firebaseUser = await admin
-      .auth()
-      .setCustomUserClaims(request.user.externalId, {
-        [request.body.role]: true,
-      })
+    const firebaseUser = await admin.auth().getUser(request.user.externalId)
+    const claims = firebaseUser.customClaims
+
+    // If the user is not admin OR the logged in user isn't the admin email, throw error
+    const isAdminUser =
+      !!claims && Object.keys(claims).includes(FirebaseClaim.admin)
```

Comment:
How about `!!claims && (FirebaseClaim.admin in claims)`? or `claims?.[FirebaseClaim.admin]`

Reply:
Good idea! Much simpler. Updated in 6b0a58d